web/src/app/modules/shared/components/form-view-container/form-view-container.component.html

```diff
@@ -0,0 +1,111 @@
+<div [formGroup]="formGroupContainer" *ngFor="let field of form?.fields; trackBy: trackByFn">
+  <ng-container [ngSwitch]="field.type">
+
+    <ng-container *ngSwitchCase="'checkbox'">
+      <clr-checkbox-container>
+        <label [for]="field.name">{{ field.label }}</label>
+        <clr-checkbox-wrapper
+          *ngFor="let opt of fieldChoices(field); trackBy: trackByFn"
+        >
+          <input
+            type="checkbox"
+            clrCheckbox
+            [formArrayName]="field.name"
+            [value]="opt.value"
+            [checked]="opt.checked"
+            (change)="onCheck($event, field.name)"
+          />
+          <label>{{ opt.label }}</label>
+        </clr-checkbox-wrapper>
+        <clr-control-error>{{ field.error }}</clr-control-error>
+      </clr-checkbox-container>
+    </ng-container>
+
+    <ng-container *ngSwitchCase="'radio'">
+      <clr-radio-container>
+        <label [for]="field.name">{{ field.label }}</label>
+        <clr-radio-wrapper
+          *ngFor="let opt of fieldChoices(field); trackBy: trackByFn"
+        >
+          <input
+            type="radio"
+            clrRadio
+            [formControlName]="field.name"
+            [value]="opt.value"
+          />
+          <label>{{ opt.label }}</label>
+        </clr-radio-wrapper>
+        <clr-control-error>{{ field.error }}</clr-control-error>
+      </clr-radio-container>
+    </ng-container>
+
+    <ng-container *ngSwitchCase="'text'">
+      <clr-input-container>
+        <label [for]="field.name">{{ field.label }}</label>
+        <input
+          clrInput
+          type="text"
+          [formControlName]="field.name"
+          [placeholder]="field.placeholder ? field.placeholder : ''"
+        />
+        <clr-control-error>{{ field.error }}</clr-control-error>
+      </clr-input-container>
+    </ng-container>
+
+    <ng-container *ngSwitchCase="'number'">
```

Comment:
If the default value is used instead of an updated value, the return type is a string. A form type number should always return a number.

Reply:
I am not completely sure what you mean but I was thinking in something like this 
`
  transformValue(field): any {
    if (field.type === 'number') {
      const value = +field.value;
      return Number.isNaN(value) ? 0 : value;
    }
  }`
this function will be called during form creation

Reply:
And if is empty string should I return 0 or nil?

Reply:
it was decided to use nil

Reply:
Done!